src/components/GraphFilter/GraphFilter.tsx

```diff
@@ -59,12 +78,13 @@ export class GraphFilter extends React.Component<GraphFilterProps, GraphFilterSt
               id="namespace-selector"
               title={this.props.activeNamespace.name}
               onSelect={this.updateNamespace}
+              onClick={this.reloadData}
             >
-              {this.state.availableNamespaces.map(ns => (
-                <MenuItem key={ns.name} active={ns.name === this.props.activeNamespace.name} eventKey={ns.name}>
-                  {ns.name}
-                </MenuItem>
-              ))}
+              <NSList
```

Comment:
Let's spell out `Namespace` its only a few more characters and is meaningful to everyone reading the code (at least in the TagName)

Reply:
ack